astropy/units/quantity.py

```diff
@@ -459,6 +459,13 @@ def __new__(
         # To ensure array remains fast, we short-circuit it.
         value_unit = None
         if not isinstance(value, np.ndarray):
+            from astropy.utils.compat.optional_deps import HAS_ARRAY_API_COMPAT
```

Comment:
I don't think we need this explicit test - this branch just tests for string and sequences of Quantity, and otherwise falls through to an `np.array` call which will already do the right thing. See main comment.